src/Workspaces/Core/Portable/Workspace/Solution/ChecksumWithChildren.cs

```diff
@@ -11,16 +11,16 @@ namespace Microsoft.CodeAnalysis.Serialization
     /// <summary>
     /// this is a collection that has its own checksum and contains only checksum or checksum collection as its children.
     /// </summary>
-    internal abstract class ChecksumWithChildren(ImmutableArray<object> children) : IChecksummedObject
+    internal abstract class ChecksumWithChildren(ImmutableArray<IChecksummedObject> children) : IChecksummedObject
     {
         public Checksum Checksum { get; } = CreateChecksum(children);
 
-        public ImmutableArray<object> Children { get; } = children;
+        public ImmutableArray<IChecksummedObject> Children { get; } = children;
 
-        private static Checksum CreateChecksum(ImmutableArray<object> children)
+        private static Checksum CreateChecksum(ImmutableArray<IChecksummedObject> children)
         {
             // given children must be either Checksum or Checksums (collection of a checksum)
-            return Checksum.Create(children.Select(c => c as Checksum ?? ((ChecksumCollection)c).Checksum));
+            return Checksum.Create(children);
```

Comment:
>Checksum.Create(children); [](http://example.com/codeflow?start=19&length=26)

Is the IEnumerable version of this still necessary?

Reply:
possibly not.  will roll into my followup.